tensorboard/java/org/tensorflow/tensorboard/vulcanize/Vulcanize.java

```diff
@@ -747,11 +746,26 @@ private static void combineScriptElements(Document document) {
       if (src.isEmpty()) {
         sourceContent = script.html();
       } else {
-        // script element that remains are the one that are annotated with `jscomp-ignore` or one
-        // that appear inside descendant of a node annotated with `vulcanize-noinline`.  They must
-        // resolve from the root because those srcs are rootified.
-        Webpath path = Webpath.get("/").resolve(Webpath.get(src)).normalize();
-        sourceContent = new String(Files.readAllBytes(webfiles.get(path)), UTF_8);
+        // script element that remains are the ones with src that is absolute, annotated with
+        // `jscomp-ignore`, or appear inside descendant of a node annotated with
+        // `vulcanize-noinline`. They must resolve from the root because those srcs are rootified.
+        Webpath webpathSrc = Webpath.get(src);
+        if (isAbsolutePath(webpathSrc)) {
+          System.err.println(
+              "WARNING: "
+                  + webpathSrc
+                  + " refers to a remote resource. Please add it to CSP manually. Detail: "
+                  + script.outerHtml());
+          continue;
+        } else if (!webfiles.containsKey(webpathSrc)) {
+          System.err.println(
+              "WARNING: source content of "
+                  + webpathSrc
+                  + " is unknown to the Vulcanization. Please add it to CSP manually. Detail: "
+                  + script.outerHtml());
+          continue;
```

Comment:
to reviewer: I can hard fail in this situation; there should never be a case when a webpath does not resolve to a webfiles. Practically, `tf_web_library` and other closure tooling would catch such case even before/after Vulcanization though.

Reply:
IMO maybe a bit better to hard fail, yeah.

I wonder if there's a way we can make the other case (remote resource) also more obviously enforceable?  I worry about people not seeing these warnings.  For example, maybe we could make that check also hard-fail unless there's a special attribute on the `<script>` tag like `manual-csp` or something?